x/market/handler/endblock_test.go

```diff
@@ -0,0 +1,133 @@
+package handler
+
+import (
+	"errors"
+	"math/rand"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/market/types"
+)
+
+var (
+	// Create and seed the generator.
+	// Typically a non-fixed seed should be used, such as time.Now().UnixNano().
+	// Using a fixed seed will produce the same output on every run.
+	r = rand.New(rand.NewSource(550000))
+)
+
+type winnerTest struct {
+	desc      string
+	bids      []types.Bid
+	expWinner *types.Bid
+	expErr    error
+}
+
+func (w *winnerTest) testFunc(t *testing.T) {
+	winner, err := pickBidWinner(w.bids)
+	if !errors.Is(err, w.expErr) {
+		t.Errorf("returned err: %v does not match %v", err, w.expErr)
+	}
+	if w.expWinner != nil && !winner.Equals(w.expWinner.ID()) {
+		t.Errorf("unexpected winner: %#v\n%q : %v", winner, types.BidIDString(winner.BidID), winner.Price)
+		t.Logf("winner: %+v coin: %s", winner.ID(), winner.Price.String())
+	}
+}
+
+func TestBidWinner(t *testing.T) {
+	seed := make([]byte, 32)
+	_, err := r.Read(seed)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ownerAddr := testutil.DeterministicAccAddress(t, seed)
+
+	bid0 := createBid(t, int64(3))
+	bid0.Owner = ownerAddr
```

Comment:
```go
  oid := testutil.OrderID(t)
  bid0 := createBid(t,oid,3)
//....

func createBid(t testing.TB, oid types.OrderID, amount int) types.Bid {
  return types.Bid{
    ID:    types.MakeBidID(oid,testutil.AccAddress(t)),
    Price: sdk.NewCoin("akash",int64(amount)),
  }  
}
```

Reply:
That's much better, I should have noticed passing the OrderID was an option.